marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -592,10 +599,931 @@ declare function hent:dump-tde($entities as json:array)
+                        <tde:scalar-type>string</tde:scalar-type>
+                        <tde:val>.</tde:val>
+                        {$is-nullable}
+                      </tde:column>
+                    else
+                      <tde:column>
+                        { comment { "This column holds array values from property",
+                        $primary-key-name, "of",
+                        $entity-type-name } }
+                        <tde:name>{ $property-name=>$secure-tde-name() }</tde:name>
+                        <tde:scalar-type>{ if ($items-datatype="iri") then "IRI" else $items-datatype }</tde:scalar-type>
+                        <tde:val>.</tde:val>
+                        {$is-nullable}
+                      </tde:column>
+                  }
+                </tde:columns>
+              </tde:row>
+            </tde:rows>
+          </tde:template>
+        )
+   return
+        (
+          if (exists($primary-key-name))
+          then
+            map:put($triples-templates, $entity-type-name,
+              <tde:template>
+                <tde:context>{ $prefix-path }{ $prefix-value }{ $entity-type-name }</tde:context>
+                <tde:vars>
+                  {
+                    if ($primary-key-type eq "string")
+                    then (
+                      <tde:var><tde:name>subject-iri</tde:name><tde:val>sem:iri(concat("{ model-graph-prefix($model) }/{ $entity-type-name }/", fn:encode-for-uri(./{ $prefix-value }{ $primary-key-name })))</tde:val></tde:var>
+                    ) else (
+                      <tde:var><tde:name>subject-iri</tde:name><tde:val>sem:iri(concat("{ model-graph-prefix($model) }/{ $entity-type-name }/", fn:encode-for-uri(xs:string(./{ $prefix-value }{ $primary-key-name }))))</tde:val></tde:var>
+                    )
+                  }
+                </tde:vars>
+                <tde:triples>
+                  <tde:triple>
+                    <tde:subject><tde:val>$subject-iri</tde:val></tde:subject>
+                    <tde:predicate><tde:val>$RDF_TYPE</tde:val></tde:predicate>
+                    <tde:object><tde:val>sem:iri("{ model-graph-prefix($model) }/{ $entity-type-name }")</tde:val></tde:object>
+                  </tde:triple>
+                  <tde:triple>
+                    <tde:subject><tde:val>$subject-iri</tde:val></tde:subject>
+                    <tde:predicate><tde:val>sem:iri("http://www.w3.org/2000/01/rdf-schema#isDefinedBy")</tde:val></tde:predicate>
+                    <tde:object><tde:val>fn:base-uri(.)</tde:val></tde:object>
+                  </tde:triple>
+
+                {
+                 for $property-name in map:keys($properties)
+                    let $is-related-entity-type := map:contains( map:get($properties, $property-name), "relatedEntityType" )
+                    let $related-entity-type := map:get( map:get($properties, $property-name), "relatedEntityType" )
```

Comment:
@prolonMarkLogic Can you adjust to also look under the "items" property for "relatedEntityType" so we catch the scenario where the relationship is many to one? 

Something like:
`let $property-info := if (map:contains(map:get($properties, $property-name), "items")) then 
                                        map:get(map:get($properties, $property-name), "items")
                                    else
                                        map:get($properties, $property-name)
  let $is-related-entity-type := map:contains($property-info, "relatedEntityType" )
  let $related-entity-type := map:get($property-info, "relatedEntityType" )
`

Reply:
@ryanjdew @aniamy Hi I created another class to put all the tde template logic together, after that I put the changes that Ryan wrote me and I tested this with the babyRegistry entity of the reference-entity-model project and it worked correctly. Let me know what do you think about these changes. thanks